src/odemis/acq/stream/_static.py

```diff
@@ -256,52 +256,55 @@ def __init__(self, name, data, *args, **kwargs):
         data = [d.getData() if isinstance(d, model.DataArrayShadow) else d for d in data]
 
         # find positions of each acquisition
-        # tuple of 2 floats -> DataArray: position on SEM -> data
-        self._sempos = {}
+        # tuple of 2 floats + 1 string) -> DataArray: position on SEM + position of polarization analyzer -> data
+        self._pos = {}
         for d in data:
+            if not model.MD_POL_MODE:
```

Comment:
shouldn't that be something like "if MD_POL_MODE not in d.metadata"?

Reply:
Instead of checking for the metadata, just read it this way:
`self._pos[d.metadata[MD_POS] + (d.metadata.get(MD_POL_MODE, None),)] = ...`
or to be more readable:
```
sempos = d.metadata[MD_POS]
pol = d.metadata.get(MD_POL_MODE, None)
self._pos[sempos + (pol,)] = ...
```